ggshield/config.py

```diff
@@ -23,139 +24,198 @@ def replace_in_keys(data: Dict, old_char: str, new_char: str) -> None:
             data[new_key] = data.pop(key)
 
 
-class Config:
-    all_policies: bool
-    api_url: str
-    exit_zero: bool
-    matches_ignore: List[IgnoredMatch]
-    paths_ignore: Set[str]
-    show_secrets: bool
-    verbose: bool
-    allow_self_signed: bool
-    max_commits_for_hook: int
-    banlisted_detectors: Set[str]
-    ignore_default_excludes: bool
-
-    CONFIG_LOCAL = ["./.gitguardian", "./.gitguardian.yml", "./.gitguardian.yaml"]
-    CONFIG_GLOBAL = [
-        os.path.join(os.path.expanduser("~"), ".gitguardian"),
-        os.path.join(os.path.expanduser("~"), ".gitguardian.yml"),
-        os.path.join(os.path.expanduser("~"), ".gitguardian.yaml"),
-    ]
-    DEFAULT_CONFIG_LOCAL = "./.gitguardian.yaml"
-
-    def __init__(self) -> None:
-        self.attributes: List[Attribute] = [
-            Attribute("all_policies", False),
-            Attribute("allow_self_signed", False),
-            Attribute("api_url", "https://api.gitguardian.com"),
-            Attribute("banlisted_detectors", set()),
-            Attribute("exit_zero", False),
-            Attribute("matches_ignore", list()),
-            Attribute("max_commits_for_hook", 50),
-            Attribute("paths_ignore", set()),
-            Attribute("show_secrets", False),
-            Attribute("verbose", False),
-            Attribute("ignore_default_excludes", False),
-        ]
-
-        for attr in self.attributes:
-            setattr(self, attr.name, attr.default)
-        self.load_configs(self.CONFIG_GLOBAL)
-        self.load_configs(self.CONFIG_LOCAL)
+def load_yaml(path: str, raise_exc: bool = False) -> Optional[Dict[str, Any]]:
```

Comment:
Maybe this function should always return an empty dict, to simplify calling code (or it should not do a fallback on empty dict when calling `safe_load()`)

Reply:
Not a blocker for me, but if I prefer using exceptions for error management than relying on the difference between `None` and `{}`: loading an empty YAML document returns `None` after all.